java/clients/src/main/java/sleeper/clients/deploy/PopulateInstanceProperties.java

```diff
@@ -42,31 +45,31 @@
 import static sleeper.configuration.properties.UserDefinedInstanceProperty.SUBNETS;
 import static sleeper.configuration.properties.UserDefinedInstanceProperty.VPC_ID;
 
-public class GenerateInstanceProperties {
+public class PopulateInstanceProperties {
     private final Supplier<String> accountSupplier;
     private final AwsRegionProvider regionProvider;
     private final String instanceId;
     private final String vpcId;
     private final String subnetIds;
-    private final Properties properties;
+    private final InstanceProperties properties;
     private final Properties tagsProperties;
 
-    private GenerateInstanceProperties(Builder builder) {
+    private PopulateInstanceProperties(Builder builder) {
         accountSupplier = requireNonNull(builder.accountSupplier, "accountSupplier must not be null");
         regionProvider = requireNonNull(builder.regionProvider, "regionProvider must not be null");
         instanceId = requireNonEmpty(builder.instanceId, "instanceId must not be empty");
         vpcId = requireNonEmpty(builder.vpcId, "vpcId must not be empty");
         subnetIds = requireNonEmpty(builder.subnetIds, "subnetIds must not be empty");
-        properties = Optional.ofNullable(builder.properties).orElseGet(Properties::new);
+        properties = Optional.ofNullable(builder.properties).orElseGet(InstanceProperties::new);
         tagsProperties = Optional.ofNullable(builder.tagsProperties).orElseGet(Properties::new);
     }
 
     public static Builder builder() {
         return new Builder();
     }
 
-    public InstanceProperties generate() {
-        InstanceProperties instanceProperties = generateDefaultsFromInstanceId(properties, instanceId);
+    public InstanceProperties populate() {
+        InstanceProperties instanceProperties = populateDefaultsFromInstanceId(properties, instanceId);
         instanceProperties.loadTags(tagsProperties);
```

Comment:
It seems awkward that DeployInstanceConfiguration puts the tags in InstanceProperties but PopulateInstanceProperties wants a separate Properties object. We could avoid overwriting the tags in the InstanceProperties if they're not explicitly provided?